include/motor-control/core/stepper_motor/motor_interrupt_handler.hpp

```diff
@@ -221,9 +221,36 @@ class MotorInterruptHandler {
         set_buffered_move(Move{});
     }
 
-    auto get_encoder_pulses() { return hardware.get_encoder_pulses(); }
+    auto get_encoder_pulses() {
+        if (get_encoder_overflow_status() == true) {
+            hardware.clear_encoder_SR();
+            if (enc_overflow_future_flag) {
+                enc_position_tracker = hardware.get_encoder_pulses();
+                enc_overflow_future_flag = false;
+            } else {
+                enc_position_tracker =
```

Comment:
I don't think this logic is correct - what happens the _second_ time there's an overflow?

Reply:
Either an overflow or a underflow can happen. When the status register is clear, the flag can get triggered again when an underflow happens. So the previous state of the overflow is saved to track the previous state because we cleared the current status register. On the head, these overflows happen only once, that's if the mounts moves greater  than ~180mm from the home position. Once that is triggered, we can detect an underflow if the motor reverses direction from less ~180mm. This logic wouldn't work if the robot triggers 2 overflows, but on the head can't trigger 2 overflows with one move. There isn't enough distance on the head to trigger 2 overflows.

Reply:
Yeah, but this is fundamental enough that I'd really like to handle that case, both so it's clearly correct code to read and to handle future physical changes. For instance, what if we put a higher-count encoder on? What if we increase the gear ratio that the encoder is on the other side of.

Reply:
I'll look into a different method to handle this.